### DISCUSSION TITLE
Top Level Folder from Document Library - PnP Modern Search

### DISCUSSION THREAD

**Discussion by kartrabby** [microsoft-search/pnp-modern-search#2200] [Q&A] [answered]
Hello, can anyone please help me with this ? I am using PnP Modern Search ( V4 ). This is how my Document library looks like

![image](https://user-images.githubusercontent.com/95188375/176666850-dc6857b6-b079-4dd9-b879-cc586d315335.png)

But the below Query Template fetches all Top Folders, subfolders and Files.

`Path:"https://you-sharepoint-site-here/Shared%20Documents/"
{searchTerms} (ParentLink:{/Documents/})`

This is the result of the above Query Template

![image](https://user-images.githubusercontent.com/95188375/176667044-e9c6bee0-cb39-4423-abb4-23f7c2f66996.png)

Please help me to get only the Top Level Folder from Document Library ( what is the Query Template ) ?

Just like this ?

![image](https://user-images.githubusercontent.com/95188375/176667106-cd87ee6a-9b65-40c5-a358-e88620a971ad.png)

**Comment by wobba** [comment, marked as answer]
Hi,
You can try to do `ParentLink="https://you-sharepoint-site-here/Shared%20Documents`"

**Reply by kartrabby** [reply]
Hi Wobba, I appreciate your response, But it's not working !

**Reply by wobba** [reply]
It was a guess on my part only. I'll see if I can come up with something, but might not be possible using search API's.

**Reply by wobba** [reply]
This seems to work: `ParentLink="https://tenant.sharepoint.com/Shared Documents/Forms/AllItems.aspx"`

**Reply by kartrabby** [reply]
Hi @wobba , Thank you so much. It did help a lot. 

Below is the Query Template Code.

```
path:https://<tenant>.sharepoint.com/sites/<site name>/
{searchTerms} ParentLink="https://<tenant>.sharepoint.com/sites/<site name>/Shared Documents/Forms/AllItems.aspx"
```

**Reply by wobba** [reply]
You can probably remove the path filter, as the ParentLink filter will override it.
